src/sentry/static/sentry/app/views/projectDetail/projectScoreCards/projectVelocityScoreCard.tsx

```diff
@@ -0,0 +1,217 @@
+import React from 'react';
+
+import AsyncComponent from 'app/components/asyncComponent';
+import {canIncludePreviousPeriod} from 'app/components/charts/utils';
+import {getParams} from 'app/components/organizations/globalSelectionHeader/getParams';
+import {parseStatsPeriod} from 'app/components/organizations/timeRangeSelector/utils';
+import ScoreCard from 'app/components/scoreCard';
+import {t, tn} from 'app/locale';
+import {GlobalSelection, Organization} from 'app/types';
+import {defined} from 'app/utils';
+import {getPeriod} from 'app/utils/getPeriod';
+
+import MissingReleasesButtons from '../missingFeatureButtons/missingReleasesButtons';
+
+const API_LIMIT = 1000;
+
+type Release = {version: string; date: string};
+
+type Props = AsyncComponent['props'] & {
+  organization: Organization;
+  selection: GlobalSelection;
+};
+
+type State = AsyncComponent['state'] & {
+  currentReleases: Release[] | null;
+  previousReleases: Release[] | null;
+  noReleaseEver: boolean;
+};
+
+class ProjectVelocityScoreCard extends AsyncComponent<Props, State> {
+  getDefaultState() {
+    return {
+      ...super.getDefaultState(),
+      currentReleases: null,
+      previousReleases: null,
+      noReleaseEver: false,
+    };
+  }
+
+  getEndpoints() {
+    const {organization, selection} = this.props;
+
+    const {projects, environments, datetime} = selection;
+    const {period} = datetime;
+    const commonQuery = {
+      environment: environments,
+      project: projects[0],
+    };
+    const endpoints: ReturnType<AsyncComponent['getEndpoints']> = [
+      [
+        'currentReleases',
+        `/organizations/${organization.slug}/releases/stats/`,
+        {
+          includeAllArgs: true,
+          method: 'GET',
+          query: {
+            ...commonQuery,
+            ...getParams(datetime),
+          },
+        },
+      ],
+    ];
+
+    if (period && canIncludePreviousPeriod(true, period)) {
+      const {start: previousStart} = parseStatsPeriod(
+        (getPeriod(
+          {period, start: undefined, end: undefined},
+          {shouldDoublePeriod: true}
+        ) as {statsPeriod: string}).statsPeriod
+      );
+
+      const {start: previousEnd} = parseStatsPeriod(
+        (getPeriod(
+          {period, start: undefined, end: undefined},
+          {shouldDoublePeriod: false}
+        ) as {statsPeriod: string}).statsPeriod
```

Comment:
Should we update the `getPeriod()` function to not return a union and instead have a consistent return shape? Having to cast each time the method is used is tedious.